connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/HerderRequestHandler.java

```diff
@@ -66,17 +87,15 @@ public <T, U> T completeOrForwardRequest(FutureCallback<T> cb,
                                                     Translator<T, U> translator,
                                                     Boolean forward) throws Throwable {
         try {
-            return cb.get(requestTimeoutMs, TimeUnit.MILLISECONDS);
-        } catch (ExecutionException e) {
-            Throwable cause = e.getCause();
-
-            if (cause instanceof RequestTargetException) {
+            return completeRequest(cb);
+        } catch (Exception e) {
+            if (e instanceof RequestTargetException) {
```

Comment:
Can we replace the `catch (Exception e)` and following `if (e instanceof ...)` blocks with dedicated `catch` blocks for each exception type?

E.g.:

```java
try {
    return completeRequest(cb);
} catch (RequestTargetException e) {
    if (forward == null || forward) {
        // ...
```